gravitee-am-management-api/gravitee-am-management-api-service/src/test/java/io/gravitee/am/management/service/impl/upgrades/DomainPasswordPoliciesUpgraderTest.java

```diff
@@ -0,0 +1,138 @@
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.am.management.service.impl.upgrades;
+
+import io.gravitee.am.model.Domain;
+import io.gravitee.am.model.PasswordSettings;
+import io.gravitee.am.model.ReferenceType;
+import io.gravitee.am.model.SystemTask;
+import io.gravitee.am.model.SystemTaskStatus;
+import io.gravitee.am.repository.management.api.SystemTaskRepository;
+import io.gravitee.am.service.DomainService;
+import io.gravitee.am.service.PasswordPolicyService;
+import io.reactivex.rxjava3.core.Flowable;
+import io.reactivex.rxjava3.core.Maybe;
+import io.reactivex.rxjava3.core.Single;
+import org.junit.jupiter.api.DisplayNameGeneration;
+import org.junit.jupiter.api.DisplayNameGenerator;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.UUID;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Eric LELEU (eric.leleu at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@ExtendWith(MockitoExtension.class)
+@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
+public class DomainPasswordPoliciesUpgraderTest {
+
+    @InjectMocks
+    private DomainPasswordPoliciesUpgrader upgrader = new DomainPasswordPoliciesUpgrader();
+
+    @Mock
+    private SystemTaskRepository systemTaskRepository;
+
+    @Mock
+    private DomainService domainService;
+
+    @Mock
+    private PasswordPolicyService passwordPolicyService;
+
+    @Test
+    public void should_ignore_if_task_completed() {
+        final SystemTask task = new SystemTask();
+        task.setStatus(SystemTaskStatus.SUCCESS.name());
+        when(systemTaskRepository.findById(any())).thenReturn(Maybe.just(task));
+
+        upgrader.upgrade();
+
+        verify(systemTaskRepository, times(1)).findById(any());
+        verify(domainService, never()).listAll();
+    }
+
+    @Test
+    public void should_ignore_domain_without_passwordSettings() {
+        initializeSystemTask();
+        var domainNoPasswordSettings = new Domain();
+
+        when(domainService.listAll()).thenReturn(Flowable.just(domainNoPasswordSettings));
```

Comment:
I recommend adding more domains here and verifying how many updates have been performed.

Reply:
I have created a dedicated test with 3 domains (2 with settings, 1 without) to test that all domains are manage properly.
I prefere keeping a test with single case in addition of this one to ease the identification of future regression